### PR TITLE
[Blazor] Finalize the integration between runtime and Blazor

### LINKED ISSUE #87178: Blazor shouldn't enable debugging when not debugging
With  .NET 8 Preview 3, I’m seeing the following dotnet.generated.invalid node show up in the sources tab of the browser dev tools when running a new Blazor WebAssembly app:
 
image

I don't see any other errors in the browser dev console.

Repro steps:

- Install .NET 8 Preview 3 SDK
- `dotnet new blazorwasm`
- `dotnet run`
- Open the browser dev tools and look at the Source tab

Expected result:
- No dotnet.generated.invalid node
- Jiterpreter generated wasm modules are shown

Actual result: 
- A dotnet.generated.invalid node shows up
- No generated wasm modules shown

### PR DESCRIPTION
**Pull request by maraf**
- It depends on 
  - https://github.com/dotnet/runtime/pull/87959
  - https://github.com/dotnet/aspnetcore/pull/49400
- Drop extra download of boot config
  - We don't have the hash of dotnet.js loader anymore
- Move lazy assembly loading to runtime
  - https://github.com/dotnet/runtime/blob/main/src/mono/wasm/runtime/lazyLoading.ts
- Move satellite assembly loading to runtime
  - https://github.com/dotnet/runtime/blob/main/src/mono/wasm/runtime/satelliteAssemblies.ts
- Move library initializers loading to runtime
  - https://github.com/dotnet/runtime/blob/main/src/mono/wasm/runtime/loader/libraryInitializers.ts
  - Call Blazor specific callbacks in Blazor using provided API
- All moved features are tested in runtime
  - https://github.com/dotnet/runtime/tree/main/src/mono/wasm/Wasm.Build.Tests/TestAppScenarios
- Update dotnet.d.ts
  - Update emscripten `out` & `err` functions

Closes https://github.com/dotnet/runtime/issues/87178

### CODE REVIEW

**Comment by maraf** [comment]
@javiercn @MackinnonBuck The CI will fail until changes from runtime flow to aspnetcore, but I would appreciate your feedback sooner, ideally to make it to preview7.

File: src/Components/Web.JS/src/Platform/Mono/MonoPlatform.ts [hunk 1]
```diff
@@ -215,86 +166,64 @@ async function importDotnetJs(startOptions: Partial<WebAssemblyStartOptions>): P
     }
   }
 
-  // For consistency with WebAssemblyResourceLoader, we only enforce SRI if caching is allowed
```
Comment (pavelsavara) [inline src/Components/Web.JS/src/Platform/Mono/MonoPlatform.ts:218]: Does runtime behave in the same way W.R.T. SRI/caching ?

File: src/Components/WebAssembly/WebAssembly/src/Services/LazyAssemblyLoader.cs [hunk 2]
```diff
@@ -18,8 +18,6 @@ namespace Microsoft.AspNetCore.Components.WebAssembly.Services;
 /// </summary>
 public sealed partial class LazyAssemblyLoader
 {
-    private HashSet<string>? _loadedAssemblyCache;
```
Comment (pavelsavara) [inline src/Components/WebAssembly/WebAssembly/src/Services/LazyAssemblyLoader.cs:21]: This was probably preventing duplicate load, is that still working ?

Comment (maraf) [reply src/Components/WebAssembly/WebAssembly/src/Services/LazyAssemblyLoader.cs:21]: It is moved to runtime for WebAssembly https://github.com/dotnet/runtime/blob/934e7e475e3ce71539d2065ae3e06d19a664b46f/src/mono/wasm/runtime/lazyLoading.ts#L20

File: src/Components/Web.JS/src/Platform/Mono/MonoPlatform.ts (hunk 1 above)
Comment (maraf) [reply src/Components/Web.JS/src/Platform/Mono/MonoPlatform.ts:218]: I don't have the hash for `dotnet.js` anymore

File: src/Components/Web.JS/src/Platform/Mono/MonoPlatform.ts [hunk 3]
```diff
@@ -150,63 +146,18 @@ export const monoPlatform: Platform = {
...
-    .filter(n => n.startsWith('dotnet.') && n.endsWith('.js'))[0];
-  const dotnetJsContentHash = bootConfig.resources.runtime[dotnetJsResourceName];
-  let src = `_framework/${dotnetJsResourceName}`;
+  let src = '_framework/dotnet.js';
 
   // Allow overriding the URI from which the dotnet.*.js file is loaded
   if (startOptions.loadBootResource) {
     const resourceType: WebAssemblyBootResourceType = 'dotnetjs';
-    const customSrc = startOptions.loadBootResource(resourceType, dotnetJsResourceName, src, dotnetJsContentHash);
+    const customSrc = startOptions.loadBootResource(resourceType, 'dotnet.js', src, '');
```
Comment (maraf) [inline src/Components/Web.JS/src/Platform/Mono/MonoPlatform.ts:160]: This is probably a breaking change, because we don't have a hash of dotnet.js anymore

**Review by pavelsavara — APPROVED** [review]

**Review by MackinnonBuck — APPROVED** [review]

**Pull request merged**